src/management/src/management_frontend.hpp

```diff
@@ -836,8 +1042,35 @@ class RequestHandler {
     return app_metadata["default_output"];
   }
 
-  bool set_model_version(const string& model_name,
-                         const string& new_model_version) {
+  /**
+   * Creates an endpoint that listens for requests to update
+   * a specified model to a specified version.
+   *
+   * JSON format:
+   * {
+   *  "model_name" := string,
+   *  "model_version" := string,
+   * }
+   */
+  std::string set_model_version(const std::string& json) {
+    rapidjson::Document d;
+    parse_json(json, d);
+    std::string model_name = get_string(d, "model_name");
+    std::string new_model_version = get_string(d, "model_version");
+
+    check_updated_model_consistent_with_app_links(model_name,
```

Comment:
The `check_updated_model_consistent_with_app_links` method is being invoked with `new_model_version` as the second parameter, as opposed to an `input_type` value.

Also, `attempt_model_version_update` calls the same method again. Does it need to be called in both places?